src/realm/obj.cpp

```diff
@@ -1439,16 +1428,18 @@ Obj& Obj::set<ObjKey>(ColKey col_key, ObjKey target_key, bool is_default)
     ColKey::Idx col_ndx = col_key.get_index();
     ColumnType type = col_key.get_type();
     if (type != ColumnTypeTraits<ObjKey>::column_id)
-        throw LogicError(LogicError::illegal_type);
+        throw InvalidArgument(ErrorCodes::TypeMismatch,
+                              util::format("Property not a %1", ColumnTypeTraits<int64_t>::column_id));
```

Comment:
This should be `ColumnTypeTraits<ObjKey>::column_id`, but even that will produce "col_type_Link" why not just hard code a more readable string message? "Property is not a link"

Reply:
This is what happens when you copy/paste